airbus_harvester/__main__.py

```diff
@@ -146,65 +159,39 @@ def harvest(workspace_name: str, catalog: str, s3_bucket: str):
 
         latest_harvested["summary"] = catalogue_data_summary
 
-        # Don't send empty or nearly empty pulsar messages
-        if (
-            len(harvested_keys["added_keys"])
-            + len(harvested_keys["updated_keys"])
-            + len(deleted_keys)
-            > minimum_message_entries
-        ):
-            # Record harvested hash data in S3
-            upload_file_s3(json.dumps(latest_harvested), s3_bucket, metadata_s3_key, s3_client)
-
-            output_data = {
-                "id": f"{workspace_name}/{config['collection_name']}_{url_count}",
-                "workspace": workspace_name,
-                "bucket_name": s3_bucket,
-                "added_keys": list(harvested_keys["added_keys"]),
-                "updated_keys": list(harvested_keys["updated_keys"]),
-                "deleted_keys": deleted_keys,
-                "source": "airbus",
-                "target": "/",
-            }
-
-            send_pulsar_message(output_data, pulsar_client)
-
-            harvested_keys = {"added_keys": set(), "updated_keys": set()}
+        if len(harvested_data.keys()) >= minimum_message_entries:
 
-    upload_file_s3(json.dumps(latest_harvested), s3_bucket, metadata_s3_key, s3_client)
+            # Send message for altered keys
+            msg = {
+                "harvested_data": harvested_data,
+                "deleted_keys": [],
+            }
+            airbus_harvester_messager.consume(msg)
+            harvested_data = {}
 
+    # Do not updated collection
     if latest_harvested.get(collection_key) == previously_harvested.get(collection_key):
-        harvested_keys["updated_keys"].discard(collection_key)
+        harvested_data.discard(collection_key)
 
     # Remove this otherwise it will be marked for deletion
     previously_harvested.pop(collection_key, None)
 
     deleted_keys = list(previously_harvested.keys())
-    output_data = {
-        "id": f"{workspace_name}/airbus_final",
-        "workspace": workspace_name,
-        "bucket_name": s3_bucket,
-        "added_keys": list(harvested_keys["added_keys"]),
-        "updated_keys": list(harvested_keys["updated_keys"]),
-        "deleted_keys": deleted_keys,
-        "source": "airbus",
-        "target": "/",
-    }
 
-    if any([harvested_keys["added_keys"], harvested_keys["updated_keys"], deleted_keys]):
-        # Send Pulsar message containing harvested links
-        send_pulsar_message(output_data, pulsar_client)
-        logging.info(f"Sent harvested message {output_data}")
-    else:
-        logging.info("No changes made to previously harvested state")
+    # Send message for altered keys
+    msg = {"harvested_data": harvested_data, "deleted_keys": deleted_keys}
+    airbus_harvester_messager.consume(msg)
 
-    return output_data
+    upload_file_s3(json.dumps(latest_harvested), s3_bucket, metadata_s3_key, s3_client)
 
 
 def send_pulsar_message(output_data: dict, pulsar_client: PulsarClient, retries: int = 0) -> None:
```

Comment:
Is this function still used? And if not, how are producer names kept unique so that there won't be any conflicts?

Reply:
Also retries, but that might be taken into account elsewhere

Reply:
Retries should be handled within the Messagers class, `send_pulsar_message` is not actually used anymore so I have removed it.